ras_backstage/controllers/party_controller.py

```diff
@@ -35,3 +35,16 @@ def get_party_by_respondent_id(party_id):
 
     logger.debug('Successfully retrieved respondent party', party_id=party_id)
     return json.loads(response.text)
+
+
+def get_businesses_by_search(query):
+    logger.debug('Retrieving businesses by search query', query=query)
+    url = f'{app.config["RAS_PARTY_SERVICE"]}party-api/v1/businesses/search'
+    response = request_handler('GET', url, auth=app.config['BASIC_AUTH'], params={"query": query})
+
+    if response.status_code != 200:
```

Comment:
I mentioned this in some other reviews today. I really think we should be using raise_for_status and handling any error codes in the > 300 range.

Reply:
In this particular case we can use raise_for_status but there are certain scenarios with error codes in the > 300 range that should not actually raise any errors because the backstage is used as an aggregator. So this is just the standard way we handle responses throughout this application.